tests/test_queuing.py

```diff
@@ -0,0 +1,385 @@
+                    assert op in three_mock_queuing_contexts[1].queue
+
+                    assert len(three_mock_queuing_contexts[2].queue) == 1
+                    assert op in three_mock_queuing_contexts[2].queue
+
+                    QueuingContext.remove(op)
+
+        assert not three_mock_queuing_contexts[0].queue
+        assert not three_mock_queuing_contexts[1].queue
+        assert not three_mock_queuing_contexts[2].queue
+
+
+class TestQueue:
+    """Test the Queue class."""
+
+    def test_arbitrary_obj(self):
+        """Tests that arbitrary objects can be appended to and removed from the queue."""
+
+        objs = [5, "hi", 1.2, np.einsum, lambda x: x + 1]
+        with qml._queuing.Queue() as q:
+            for obj in objs:
+                q.append(obj)
+        assert q.queue == objs
+
+        with q:
+            for _ in range(len(objs)):
+                obj = objs.pop()
+                q.remove(obj)
+                assert q.queue == objs
+
+    def test_append_qubit_gates(self):
+        """Test that gates are successfully appended to the queue."""
+        with qml._queuing.Queue() as q:
+            ops = [
+                qml.RX(0.5, wires=0),
+                qml.RY(-10.1, wires=1),
+                qml.CNOT(wires=[0, 1]),
+                qml.PhaseShift(-1.1, wires=18),
+                qml.T(wires=99),
+            ]
+        assert q.queue == ops
+
+    def test_append_qubit_observables(self):
+        """Test that ops that are also observables are successfully
+        appended to the queue."""
+        with qml._queuing.Queue() as q:
+            # wire repetition is deliberate, Queue contains no checks/logic
+            # for circuits
+            ops = [
+                qml.Hadamard(wires=0),
+                qml.PauliX(wires=1),
+                qml.PauliY(wires=1),
+                qml.Hermitian(np.ones([2, 2]), wires=7),
+            ]
+        assert q.queue == ops
+
+    def test_append_tensor_ops(self):
+        """Test that ops which are used as inputs to `Tensor`
+        are successfully added to the queue, but no `Tensor` object is."""
+
+        with qml._queuing.Queue() as q:
+            A = qml.PauliZ(0)
+            B = qml.PauliY(1)
+            tensor_op = qml.operation.Tensor(A, B)
+        assert q.queue == [A, B]
+        assert tensor_op.obs == [A, B]
+        assert all(not isinstance(op, qml.operation.Tensor) for op in q.queue)
+
+    def test_append_tensor_ops_overloaded(self):
+        """Test that Tensor ops created using `@`
+        are successfully added to the queue, but no `Tensor` object is."""
+
+        with qml._queuing.Queue() as q:
+            A = qml.PauliZ(0)
+            B = qml.PauliY(1)
+            tensor_op = A @ B
+        assert q.queue == [A, B]
+        assert tensor_op.obs == [A, B]
+        assert all(not isinstance(op, qml.operation.Tensor) for op in q.queue)
+
+
+class TestOperationRecorder:
```

Comment:
Tests moved here from `test_utils.py`. Other than renaming of methods, nothing has changed